src/main/java/org/opensearch/ad/rest/RestValidateAnomalyDetectorAction.java

```diff
@@ -117,33 +150,89 @@ protected BaseRestHandler.RestChannelConsumer prepareRequest(RestRequest request
             }
         }
 
-        return channel -> {
-            AnomalyDetector detector;
-            try {
-                detector = AnomalyDetector.parse(parser);
-            } catch (Exception ex) {
-                if (ex instanceof ADValidationException) {
-                    ADValidationException ADException = (ADValidationException) ex;
-                    DetectorValidationIssue issue = new DetectorValidationIssue(
-                        ADException.getAspect(),
-                        ADException.getType(),
-                        ADException.getMessage()
-                    );
-                    sendAnomalyDetectorValidationParseResponse(issue, channel);
-                    return;
-                } else {
-                    throw ex;
-                }
+        AnomalyDetector detector;
+        try {
+            detector = AnomalyDetector.parse(parser);
+        } catch (Exception ex) {
+            if (ex instanceof ADValidationException) {
+                ADValidationException ADException = (ADValidationException) ex;
+                DetectorValidationIssue issue = new DetectorValidationIssue(
+                    ADException.getAspect(),
+                    ADException.getType(),
+                    ADException.getMessage()
+                );
+                return new ExtensionRestResponse(
+                    request,
+                    RestStatus.OK,
+                    new ValidateAnomalyDetectorResponse(issue).toXContent(JsonXContent.contentBuilder())
+                );
```

Comment:
Can we move line 164-168 under `sendAnomalyDetectorValidationParseResponse` method?

Reply:
Uh, lines 164-168 are the replacement for `sendAnomalyDetectorValidationParseResponse` which should probably be deleted.

Unless by "under" you mean, change this to `return sendAnomalyDetectorValidationParseResponse(issue)` and move the 3 lines into that method?  Sure.